dashboard/app/models/user.rb

```diff
@@ -2261,6 +2259,40 @@ def providers
     end
   end
 
+  def num_failed_section_attempts
+    properties['section_attempts'] || 0
+  end
+
+  def reset_section_attempts?
```

Comment:
I think a header comment on this function describing the reset logic would be useful.